libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxDistanceFormatter.kt

```diff
@@ -81,51 +76,40 @@ class MapboxDistanceFormatter(
      * relative size of .65 times the size of the number
      */
     override fun formatDistance(distance: Double): SpannableString {
-        val distanceSmallUnit =
-            TurfConversion.convertLength(distance, UNIT_METERS, smallUnit)
-        val distanceLargeUnit =
-            TurfConversion.convertLength(distance, UNIT_METERS, largeUnit)
-
-        return when {
-            // If the distance is greater than 10 miles/kilometers, then round to nearest mile/kilometer
-            distanceLargeUnit > LARGE_UNIT_THRESHOLD -> {
-                getDistanceString(roundToDecimalPlace(distanceLargeUnit, 0), largeUnit)
+        val distanceAndSuffix = when (distance) {
+            in 0.0..smallDistanceUpperThresholdInMeters -> {
+                formatDistanceAndSuffixForSmallUnit(distance)
             }
-            // If the distance is less than 401 feet/meters, round by fifty feet/meters
-            distanceSmallUnit < SMALL_UNIT_THRESHOLD -> {
-                getDistanceString(roundToClosestIncrement(distanceSmallUnit), smallUnit)
+            in mediumDistanceLowerThresholdInMeters..mediumDistanceUpperThresholdInMeters -> {
```

Comment:
what if distance equals `400.5`?. Need tests for it 
Also cold use following ranges 
```
companion object {
  DISTANCE_THRESHOLD_BOTTOM_BORDER = 400.0
  DISTANCE_THRESHOLD_UPPER_BORDER = 10000.0
}
override fun formatDistance(distance: Double): SpannableString {
  val distanceAndSuffix = when (distance) { 
  0.0..DISTANCE_THRESHOLD_BOTTOM_BORDER -> {
     ....
  }
 DISTANCE_THRESHOLD_BOTTOM_BORDER..DISTANCE_THRESHOLD_UPPER_BORDER -> {
     ....
  }
```

Reply:
Good catch, test added.